packages/dataviews/src/dataviews-layouts/hooks/use-clickable-item-props.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+export const useClickableItemProps = < Item >(
+	item: Item,
+	isItemClickable: ( item: Item ) => boolean,
+	onClickItem: ( item: Item ) => void,
+	className: string
+) => {
+	const isClickable = isItemClickable( item );
+
+	return {
+		className: clsx( className, {
+			[ className + '--clickable' ]: isClickable,
+		} ),
+		role: isClickable ? 'button' : undefined,
+		tabIndex: isClickable ? 0 : undefined,
+		onClick: ! isClickable ? undefined : () => onClickItem( item ),
+		onKeyDown: ! isClickable
+			? undefined
+			: ( event: React.KeyboardEvent ) => {
+					if ( event.key === 'Enter' || event.key === '' ) {
+						onClickItem( item );
+					}
+			  },
+	};
```

Comment:
This seems needlessly complicated. If the item isn't clickable, all attributes but class will be empty. So:

```suggestion
	if ( ! isItemClickable( item ) ) {
		return { className };
	}

	return {
		className: `${ className } ${ className }--clickable`,
		role: 'button',
		tabIndex: 0,
		onClick: () => onClickItem( item ),
		onKeyDown: ( event: React.KeyboardEvent ) => {
			if ( event.key === 'Enter' || event.key === '' ) {
				onClickItem( item );
			}
		},
	};
```

Also, see my other comment about `&--clickable`. If it turns out we don't need that class, then this could be further simplified by not handling `className` altogether. The caller would handle its own class more explicitly:

```js
const clickableProps = useClickableItemProps(
	item,
	isItemClickableField,
	onClickItem
);

return (
	<div
		className="dataviews-view-table__cell-content"
		{ ...clickableProps }
```

Reply:
Addressed at https://github.com/WordPress/gutenberg/pull/66365/commits/f24172ba41318288889fb46fe1689f90fbdc74ff